tests/integration/operator-integ-suite-ocp.sh

```diff
@@ -24,14 +24,47 @@ set -u
 # Print commands
 set -x
 
-# deploy operator in OCP
-echo "--------------------------------"
-echo "Check that istio operator is running in OCP"
-echo "--------------------------------"
-"${WD}"/check-operator.sh
-
-# create a Istio Helm Install CP and test httpbin
-echo "--------------------------------"
-echo "Create a IstioHelmInstall Control Plane and test httpbin"
-echo "--------------------------------"
-"${WD}"/control-plane-test.sh
+OPERATOR_NS="${OPERATOR_NS:-istio-operator}"
+OPERATOR_NAME="${OPERATOR_NAME:-istio-operator}"
+CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-default}"
+
+BRANCH="${BRANCH:-maistra-3.0}"
+ISTIO_HELM_INSTALL="https://raw.githubusercontent.com/maistra/istio-operator/${BRANCH}/config/samples/maistra.io_v1_istiohelminstall.yaml"
+
+TIMEOUT="3m"
+
+check_ready() {
+    local NS=$1
+    local POD_NAME=$2
+    local DEPLOYMENT_NAME=$3
+
+    echo "Check POD: NAME SPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
+    timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
+    "until oc get pod --field-selector=status.phase=Running -n ${NS} | grep ${POD_NAME}; do sleep 5; done"
+
+    if [ ${#DEPLOYMENT_NAME} -gt 0 ]; then
+      echo "Check Deployment Available: NAME SPACE: \"${NS}\"   DEPLOYMENT NAME: \"${DEPLOYMENT_NAME}\""
+      oc wait deployment "${DEPLOYMENT_NAME}" -n "${NS}" --for condition=Available=True --timeout=${TIMEOUT}
+    fi
+}
+
+
+# Main
+
+echo "Check that istio operator is running"
+check_ready "${OPERATOR_NS}" "${OPERATOR_NAME}" "${OPERATOR_NAME}"
+
+
+echo "Create a IstioHelmInstall Control Plane"
+oc get ns "${CONTROL_PLANE_NS}" >/dev/null 2>&1 || oc create namespace "${CONTROL_PLANE_NS}"
+oc apply -f "${ISTIO_HELM_INSTALL}" -n "${CONTROL_PLANE_NS}"
+
+
+echo "Check that IstioHelmInstall Control Plane is running"
+check_ready "${CONTROL_PLANE_NS}" "istiod" "istiod"
+check_ready "${CONTROL_PLANE_NS}" "istio-egressgateway" "istio-egressgateway"
+check_ready "${CONTROL_PLANE_NS}" "istio-ingressgateway" "istio-ingressgateway"
+
+
+echo "Check that CNI processes are running"
+check_ready "${OPERATOR_NS}" "istio-cni" ""
```

Comment:
istio-cni uses a daemonset, not deployment.

Reply:
Now checking daemonset rollout status.